### PR TITLE
[xmlsec] Update to version 1.3.6

### PR DESCRIPTION
**Pull request by metsma**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [X] SHA512s are updated for each updated download.
- [ ] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

Signed-off-by: Raul Metsma <raul@metsma.ee>

### CODE REVIEW

File: ports/xmlsec/CMakeLists.txt [hunk 1]
```diff
@@ -106,17 +103,19 @@ install(TARGETS xmlsec1 xmlsec1-openssl
     ARCHIVE DESTINATION lib
 )
 
+set(CMAKE_EXPERIMENTAL_EXPORT_PACKAGE_DEPENDENCIES 1942b4fa-b2c5-4546-9385-83f254070067)
```
Comment (dg0yt) [inline ports/xmlsec/CMakeLists.txt:106]: IMO experimental features should not be used in a portfile. The portfile must work with future versions of CMake, but the experimental feature might be changed in incompatible ways.

**Review by dg0yt — COMMENTED** [review]
The `LIBXML_STATIC` definitions should be removed. This is handled by the libxml2 port and has not effect in vcpkg.

File: ports/xmlsec/CMakeLists.txt (hunk 1 above)
Comment (metsma) [reply ports/xmlsec/CMakeLists.txt]: Removed and added instructions to usage file

File: ports/xmlsec/CMakeLists.txt [hunk 2]
```diff
@@ -76,28 +74,25 @@ add_compile_definitions(HAVE_CTYPE_H)
...
 
+set(XMLSEC_CORE_CFLAGS "")
 if(NOT BUILD_SHARED_LIBS)
-    set(XMLSEC_CORE_CFLAGS "-DLIBXML_STATIC -DLIBXSLT_STATIC -DXMLSEC_STATIC -DXMLSEC_NO_CRYPTO_DYNAMIC_LOADING")
-    set(XMLSEC_OPENSSL_CFLAGS ${XMLSEC_CORE_CFLAGS})
-else()
-    set(XMLSEC_CORE_CFLAGS "-DXMLSEC_DL_WIN32")
-    set(XMLSEC_OPENSSL_CFLAGS ${XMLSEC_CORE_CFLAGS})
+    set(XMLSEC_CORE_CFLAGS "-DLIBXML_STATIC -DXMLSEC_STATIC -DXMLSEC_NO_CRYPTO_DYNAMIC_LOADING")
+    target_compile_definitions(xmlsec1 PUBLIC LIBXML_STATIC XMLSEC_STATIC XMLSEC_NO_CRYPTO_DYNAMIC_LOADING)
```
Comment (dg0yt) [inline ports/xmlsec/CMakeLists.txt:87]:
```suggestion
    target_compile_definitions(xmlsec1 PUBLIC XMLSEC_STATIC XMLSEC_NO_CRYPTO_DYNAMIC_LOADING)
```
`XMLSEC_STATIC` depends on actual configuration of `BUILD_SHARED_LIBS` for libxml2, not on `BUILD_SHARED_LIBS` in xmlsec.

File: ports/xmlsec/usage [hunk 3]
```diff
@@ -10,6 +10,8 @@ xmlsec can be imported via CMake FindPkgConfig module:
 
 vcpkg provides proprietary CMake targets:
 
+    find_package(OpenSSL REQUIRED)
+    find_package(LibXml2 REQUIRED)
```
Comment (dg0yt) [inline ports/xmlsec/usage:14]: No. Please provide the `find_dependency` calls via `unofficial-xmlsec-config.cmake`.

**Review by dg0yt — CHANGES_REQUESTED** [review]

File: ports/xmlsec/usage (hunk 3 above)
Comment (metsma) [reply ports/xmlsec/usage:14]: Do you have any hints how it can be done?

File: ports/xmlsec/CMakeLists.txt (hunk 2 above)
Comment (metsma) [reply ports/xmlsec/CMakeLists.txt]: Fixed thank you

File: ports/xmlsec/usage (hunk 3 above)
Comment (metsma) [reply ports/xmlsec/usage]: Thank you fixed. Is there typo in libxmlmm config file
https://github.com/microsoft/vcpkg/blob/master/ports/libxmlmm/unofficial-libxmlmm-config.cmake#L3
It should probably include unofficial-libxmlmm-targets not unofficial-libxmlmm-config?

Comment (dg0yt) [reply ports/xmlsec/usage]: > It should probably include unofficial-libxmlmm-targets not unofficial-libxmlmm-config?

Oops. Now you got me 🤦. No downstreams unless I write a test port...

**Review by FrankXie05 — DISMISSED** [review]

File: ports/xmlsec/CMakeLists.txt [hunk 4]
```diff
@@ -60,12 +60,10 @@ set(CMAKE_STATIC_LIBRARY_PREFIX "lib")
 add_library(xmlsec1 ${SOURCESXMLSEC})
 add_library(xmlsec1-openssl ${SOURCESXMLSECOPENSSL})
 
-include_directories(${CMAKE_CURRENT_BINARY_DIR}/include include ${LIBXML2_INCLUDE_DIRS})
+include_directories(${CMAKE_CURRENT_BINARY_DIR}/include include)
```
Comment (dg0yt) [inline ports/xmlsec/CMakeLists.txt:63]: The exported targets still lack interface include directories. Untested:
```suggestion
include_directories(${CMAKE_CURRENT_BINARY_DIR}/include include)
target_include_directories(xmlsec1 PUBLIC
    $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
    include
)
```

Comment (metsma) [reply ports/xmlsec/CMakeLists.txt]: Thank you, fixed

**Review by FrankXie05 — APPROVED** [review]

**Pull request merged**